Mathlib/Data/Int/Bitwise.lean

```diff
@@ -539,7 +539,7 @@ theorem shiftLeft_eq_mul_pow : ∀ (m : ℤ) (n : ℕ), m <<< (n : ℤ) = m * (2
 theorem shiftRight_eq_div_pow : ∀ (m : ℤ) (n : ℕ), m >>> (n : ℤ) = m / (2 ^ n : ℕ)
   | (m : ℕ), n => by rw [shiftRight_coe_nat, Nat.shiftRight_eq_div_pow _ _]; simp
   | -[m+1], n => by
-    rw [shiftRight_negSucc, negSucc_ediv, Nat.shiftRight_eq_div_pow]; rfl
+    rw [shiftRight_negSucc, negSucc_ediv, Nat.shiftRight_eq_div_pow]; · rfl
```

Comment:
I'd like to better understand where the `rfl` goal comes from: is it a side goal of an rw, or a second independent goal?

Reply:
`negSucc_ediv` creates a side-goal of `0 < ↑(2 ^ n)` and `rfl` closes the main goal (which is `-[m / 2 ^ n+1] = -(div ↑m ↑(2 ^ n) + 1)`).